app/services/billable_metrics/prorated_aggregations/sum_service.rb

```diff
@@ -87,6 +97,20 @@ def duration_ratio_sql(from, to)
 
         "((DATE(#{to_in_timezone}) - DATE(#{from_in_timezone}))::numeric + 1) / #{period_duration}::numeric"
       end
+
+      def recurring_value
+        previous_charge_fee_units = previous_charge_fee&.units
+
+        return previous_charge_fee_units if previous_charge_fee_units
+
+        query = persisted_query
+          .select("SUM((CAST(#{sanitized_field_name} AS FLOAT)) * (#{persisted_pro_rata}))::numeric").to_sql
+        recurring_value_before_first_fee = ActiveRecord::Base.connection.execute(query).first['sum']
+
+        return nil unless recurring_value_before_first_fee
+
+        recurring_value_before_first_fee <= 0 ? nil : recurring_value_before_first_fee
```

Comment:
Style/TernaryParentheses: Use parentheses for ternary expressions with complex conditions.